py/tests/test_symcache.py

```diff
@@ -18,6 +18,54 @@ def test_basic(res_path):
     SymCache.from_bytes(stream.read())
 
 
+def test_symbolicate_electron_darwin_dsym(res_path):
+    path = os.path.join(res_path, 'electron/1.8.1/Electron/CB63147AC9DC308B8CA1EE92A5042E8E0/Electron.app.dSYM/Contents/Resources/DWARF/Electron')
+    fo = FatObject.from_path(path)
+    o = fo.get_object(arch='x86_64')
+    sc = o.make_symcache()
+
+    # Make sure our stream starts with the header
+    stream = sc.open_stream()
+    assert stream.read(4) == b'SYMC'
+
+    # Make s symcache from the entire thing
+    stream = sc.open_stream()
+    cache = SymCache.from_bytes(stream.read())
+
+    # Verify a known symbol
+    symbol = cache.lookup(0x107bb9f25 - 0x107bb9000)[0]
+    assert symbol.symbol == 'main'
+    assert symbol.lang == 'cpp'
+    assert symbol.line == 186
+    assert symbol.comp_dir == '/Users/electron/workspace/electron-osx-x64/out/R'
+    assert symbol.base_dir == '../../atom/app'
+    assert symbol.filename == 'atom_main.cc'
+
+
+def test_symbolicate_electron_darwin_sym(res_path):
+    path = os.path.join(res_path, 'electron/1.8.1/Electron/CB63147AC9DC308B8CA1EE92A5042E8E0/Electron.sym')
+    fo = FatObject.from_path(path)
+    o = fo.get_object(arch='x86_64')
+    sc = o.make_symcache()
+
+    # Make sure our stream starts with the header
+    stream = sc.open_stream()
+    assert stream.read(4) == b'SYMC'
+
+    # Make s symcache from the entire thing
+    stream = sc.open_stream()
+    cache = SymCache.from_bytes(stream.read())
+
+    # Verify a known symbol
+    symbol = cache.lookup(0x107bb9f25 - 0x107bb9000)[0]
+    assert symbol.symbol == 'main'
+    assert symbol.lang == 'unknown'
+    assert symbol.line == 186
+    # Comprises "comp_dir" and "base_dir"
+    assert symbol.filename == '/Users/electron/workspace/electron-osx-x64/out/R/../../atom/app/atom_main.cc'
+    # "lang", "comp_dir" and "base_dir" are not available in .sym files
```

Comment:
This could be an interesting detail. Breakpad symbols do not contain the compilation directory. If we strictly need it, some sort of heuristics would be necessary to "guess" it.